eZ/Publish/Core/Persistence/Cache/UserHandler.php

```diff
@@ -495,6 +495,30 @@ function () use ($roleId) {
         );
     }
 
+    public function loadRoleAssignmentsByRoleIdWithOffsetAndLimit(int $roleId, int $offset, int $limit): array
+    {
+        $this->logger->logCall(__METHOD__, [
+            'userId' => $roleId,
+            'offset' => $offset,
+            'limit' => $limit,
+        ]);
+
+        return $this->persistenceHandler
+            ->userHandler()
+            ->loadRoleAssignmentsByRoleIdWithOffsetAndLimit($roleId, $offset, $limit);
+    }
+
+    public function countRoleAssignments(int $roleId): int
+    {
+        $this->logger->logCall(__METHOD__, [
+            'userId' => $roleId,
+        ]);
+
+        return $this->persistenceHandler
+            ->userHandler()
+            ->countRoleAssignments($roleId);
+    }
```

Comment:
These are good candidates to be cached. Otherwise in place of 1 performance issue we're introducing another one.

Reply:
Do we want to cache role assignments with `offset` and `limit`  keys?

Reply:
@alongosz Done in https://github.com/ezsystems/ezplatform-kernel/pull/373/commits/518bcbbf97a7f2691874f4cd8aa41b22cb920294. 

However, I feel like caching count may be not optimal. Essentially, for it to work properly we need to tag count with every assignment so the tags made for single count may be huge in number (https://github.com/ezsystems/ezplatform-kernel/pull/373/files#diff-8ccb435bd05acfd0a861eabbff63986e588674045a205a0f00c8a46c0a74ff48R551).

Reply:
Conclusion on that matter after Internal sync with @adamwojs, @barw4, and @webhdx:

we can't cache count without rewriting user group view in AdminUI because it trashes groups like normal Content instead of using dedicated `UserService::deleteUserGroup` API. Rewriting it as a part of this issue is too complex, needs to be planned most likely after 4.6.0 LTS, even if we decide to fix it for 3.3.x.